tools/FlatFlowMigrationCli/SubscriptionMigrator.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using FlatFlowMigrationCli.Operations;
+using Microsoft.DotNet.ProductConstructionService.Client;
+using Microsoft.DotNet.ProductConstructionService.Client.Models;
+using Microsoft.Extensions.Logging;
+
+namespace FlatFlowMigrationCli;
+
+/// <summary>
+/// Class that performs actual write changes in subscriptions.
+/// </summary>
+internal class SubscriptionMigrator : ISubscriptionMigrator
+{
+    private readonly IProductConstructionServiceApi _pcsClient;
+    private readonly ILogger<MigrateOperation> _logger;
+
+    public SubscriptionMigrator(IProductConstructionServiceApi pcsClient, ILogger<MigrateOperation> logger)
+    {
+        _pcsClient = pcsClient;
+        _logger = logger;
+    }
+
+    public async Task DisableSubscriptionAsync(Subscription incoming)
+    {
+        var disabledSubscription = new SubscriptionUpdate
+        {
+            ChannelName = incoming.Channel.Name,
+            SourceRepository = incoming.SourceRepository,
+            Enabled = false,
+            Policy = incoming.Policy,
+            PullRequestFailureNotificationTags = incoming.PullRequestFailureNotificationTags,
+        };
+
+        await _pcsClient.Subscriptions.UpdateSubscriptionAsync(incoming.Id, disabledSubscription);
+        _logger.LogInformation("Disabled subscription {subscriptionId} {sourceRepository} -> {targetRepository}",
+            incoming.Id,
+            incoming.SourceRepository,
+            incoming.TargetRepository);
+    }
+
+    public async Task DeleteSubscriptionAsync(Subscription incoming)
+    {
+        _logger.LogInformation("Deleting an existing subscription to VMR {subscriptionId}...", incoming.Id);
+        await _pcsClient.Subscriptions.DeleteSubscriptionAsync(incoming.Id);
+    }
+
+    public async Task CreateVmrSubscriptionAsync(Subscription outgoing)
```

Comment:
maybe this method should be `CreateForwardFlowSubscriptionAsync` to be consistent with the one bellow

Reply:
looks like this method also creates backflow subscriptions tho? Isn't that wrong

Reply:
It's not a backflow, it's just a regular VMR subscription